read_buffer/src/row_group.rs

```
@@ -1186,6 +1197,30 @@ impl Ord for GroupKey<'_> {
     }
 }
 
+#[derive(PartialEq, Clone)]
+pub struct AggregateResults<'row_group>(Vec<AggregateResult<'row_group>>);
+
+impl<'row_group> AggregateResults<'row_group> {
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    fn merge(&mut self, other: &AggregateResults<'row_group>) {
+        for (i, agg) in self.0.iter_mut().enumerate() {
```

Perhaps worth an `assert_eq!(self.0.len(), other.len())`?

With `other` longer than `self.0` it'd be a silent failure otherwise.

---

In 3da9b734